lib/probe-defaults.js

```diff
@@ -0,0 +1,234 @@
+  },
+
+  // https://npmjs.org/package/amqplib
+  amqplib: {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/cassandra-driver
+  'cassandra-driver': {
+    enabled: true,
+    collectBacktraces: true,
+    sanitizeSql: true
+  },
+
+  // https://npmjs.org/package/co-render
+  'co-render': {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/director
+  director: {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/express
+  express: {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/hapi
+  hapi: {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/koa
+  koa: {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/koa-resource-router
+  'koa-resource-router': {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/koa-route
+  'koa-route': {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/koa-router
+  'koa-router': {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/levelup
+  levelup: {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/memcached
+  memcached: {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/mongodb
+  mongodb: {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/mongodb-core
+  'mongodb-core': {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/mysql
+  mysql: {
+    enabled: true,
+    collectBacktraces: true,
+    sanitizeSql: true
+  },
+
+  // https://npmjs.org/package/node-cassandra-cql
+  'node-cassandra-cql': {
+    enabled: true,
+    collectBacktraces: true,
+    sanitizeSql: true
+  },
+
+  // https://npmjs.org/package/oracledb
+  oracledb: {
+    enabled: true,
+    collectBacktraces: true,
+    sanitizeSql: true
+  },
+
+  // https://npmjs.org/package/pg
+  pg: {
+    enabled: true,
+    collectBacktraces: true,
+    sanitizeSql: true
+  },
+
+  // https://npmjs.org/package/raw-body
+  'raw-body': {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/redis
+  redis: {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/restify
+  restify: {
+    enabled: true,
+    collectBacktraces: true
+  },
+
+  // https://npmjs.org/package/tedious
+  tedious: {
+    enabled: true,
+    collectBacktraces: true,
+    sanitizeSql: true
+  },
+
+  // https://npmjs.org/package/vision
+  vision: {
+    enabled: true,
+    collectBacktraces: true,
+  },
+
+  //
+  // context maintaining modules. they cannot be disabled and are listed
+  // for completeness.
+  //
+  'aws-sdk': {enabled: true},
+  bcrypt: {enabled: true},
+  bluebird: {enabled: true},
```

Comment:
do the bluebird, generic-pool and q instrumentations need a check for enabled as well?

Reply:
no, they don't. but you spotted an inconsistency that i decided to skip over - aws-sdk is not really in the same category as the others but i didn't want to create a separate category for it.

aws-sdk was created to fix apex learning's problem. when there was an error on the http rest call to aws the aws code would resend the request but would not recalculate the signature. but the probe didn't know it was a resend and so attached a different x-trace header to the request causing it to fail the integrity check at the aws end. so, yes, that probe can be disabled but 1) it's not really a probe, it's just adds 'x-trace' to the headers that aws-sdk ignores and 2) even though it shouldn't be disabled i thought it a good idea to allow a quick disable in case there was a problem with aws-sdk at some point in the future - it would be a quick diagnostic.

the others simply maintain context which needs to be done; disabling them would cause many traces to break because they'd just lose content. again, they're not really probes. it could be argued that generic-pool should become a probe as the queue time might be interesting. but i'll cross that bridge later.